### PR TITLE
fix: for-in loop is missing hasOwnProperty check

### PR DESCRIPTION
**Pull request by LFB**
Inspection info checks for unfiltered for-in loops, The use of this construct results in processing inherited or unexpected properties.

### CODE REVIEW

File: lib/response.js [hunk 1]
```diff
@@ -463,7 +463,9 @@ module.exports = {
       this.res.setHeader(field, val);
     } else {
       for (const key in field) {
-        this.set(key, field[key]);
+        if (Object.prototype.hasOwnProperty.call(field, key)) {
```
Comment (dead-horse) [inline lib/response.js]: It can be a breaking change, why you need to add this check?

Comment (LFB) [reply lib/response.js]: Hello, the reason I want to modify is: checks for unfiltered for-in loops, The use of this construct results in processing inherited or unexpected properties. Make the code more robust.

Comment (dead-horse) [reply lib/response.js]: Sorry, if there is no actual scene, we will not modify it here, because it runs the risk of breaking other people's code

Comment (LFB) [reply lib/response.js]: Got it, thank you

**Pull request closed without merging**